src/games/strategy/triplea/delegate/StrategicBombingRaidBattle.java

```diff
@@ -203,7 +203,16 @@ public void fight(final IDelegateBridge bridge) {
     showBattle(bridge);
     final List<IExecutable> steps = new ArrayList<>();
     if (hasAA) {
-      steps.add(new FireAA());
+    	boolean foundGenericAA = false;
+      for (final Unit target : m_targets.keySet()) {
+    		//for( Unit attacker
+    		if( ( (UnitAttachment) target.getType().getAttachment("unitAttachment")).getIsAAforBombingThisUnitOnly() ) {
```

Comment:
This class is aware of the target, it knows it has types, which attachment values it has, the class type of that, and finally the methods available. If any of that changes, this class which will be off in left field will also need to change. This is poor cohesion as an update in one component causes updates in other components located throughout the system.

I recommend adding a method toUnit, so that instead of:
`(UnitAttachment) target.getType().getAttachment("unitAttachment")).getIsAAforBombingThisUnitOnly() ) {`
You would have simply: `if(unit.getIsAAForBombingThisUnitOnly()) {`